### ISSUE TITLE
Is there a way to always run local_resource first before docker_build

### ISSUE THREAD

**Issue by swingbach** [windmilleng/tilt#3319]
Hi
Firstly I'd like to say Tilt is pretty cool as a local development tool.
I tried some basic workflows, seems the execution sequence of local_resource is a bit strange to me. if I define my Tiltfile as:
```
local_resouce('test', cmd='test.sh', auto_init=True)
docker_build(...)
```
I expected not only for the first time of starting up, but each time I changed something in the build context, the test.sh script will always run first before the docker_build step start, but actually it's opposite.

Just curious is there anyway what's written in Tiltfile can be executed as the same sequence as I declared?

**Comment by victorwuky** [comment]
Thanks @swingbach for the question. Apologize for responding late. @nicks : What should be done here? Using deps somehow in local_resource?

**Comment by swingbach** [comment]
> Thanks @swingbach for the question. Apologize for responding late. @nicks : What should be done here? Using deps somehow in local_resource?

I tried deps also, it also won't really execute the local resource before the docker_build, but what's weird is local resource will be run first when you first run tilt up, but later when changes detected, local resource won't be run sometimes or at least won't be run in the same sequence as first run tilt up.

**Comment by zkutasi** [comment]
Hi,

I have followed the Java examples for the Live upgrade on the site, and there the Doc uses a trick... it makes the dockerbuild depend on something that the local_resource is changing... so you need to first do your thing with the local resource and also touch some OTHER files/folders and that will trigger the docker_build. Works like a charm with a sync command in the example Java project.

**Comment by nicks** [comment]
No, there's not currently a way to do this. See discussion here: https://github.com/windmilleng/tilt/issues/3048

After the initial startup, there are no real guarantees around build order. Ya, the Java examples use some tricks. They're *very* careful about which files they watch, and they write files to "trick" Tilt into doing things in order (e.g., writing a file X as a hack to trigger build Y).

**Comment by CpuID** [comment]
this would be awesome to see resolved btw 👍
